### PR TITLE
Test Added 

### LINKED ISSUE #18: Write Unit-Test For POST Opportunity API using Chai, Mocha and Sinon
- The POST Opportunity API is divided into controllers, services, and managers, 
Tests should be written for each 3 of them. 
- Data should be mocked with Sinon. 
- This is the first test, so set up for chai, mocha also needs to be done. 
- Refer to [this](https://blog.logrocket.com/unit-testing-node-js-applications-using-mocha-chai-and-sinon/) for help

### PR DESCRIPTION
**Pull request by Sukriti-sood**
# Description

Added Tests for Controllers, Manager, and Services

Fixes #18

### CODE REVIEW

File: test/controller-test.js
Comment (Manvityagi) [inline test/controller-test.js]: @Sukriti-sood
Rename this folder to `tests`. 
Inside tests - Make 3 folders - `controllers` , `managers`, `services` 
Inside these folders - Make file - `opportunity-test.js`

Reason for doing this: 
By current configuration, you made a file `controller-test.js`, but in the future, there will be USER APIs(and maybe more) added as well, so the structure should be a generic one so that Inside `tests->controllers`, We can have two files -`opportunity-test.js` and `user-test.js`.

**Review by Manvityagi — CHANGES_REQUESTED** [review]

File: test/controller-test.js
Comment (Sukriti-sood) [reply test/controller-test.js:1]: @Manvityagi I have done that but having some confusion regarding path for test in package.json

**Comment by Sukriti-sood** [comment]
@Manvityagi  Added Requested Changes , Please have a look  .

File: tests/managers/opportunity-test.js
Comment (Manvityagi) [inline tests/managers/opportunity-test.js]: Spelling of `OpprtunityManager` to be `OpportunityManager`

File: tests/services/opportunity-test.js
Comment (Manvityagi) [inline tests/services/opportunity-test.js]: We should follow one convention at all places, 
Let's keep the class name starting with an uppercase letter and object name starting with lowercase letter. 
So, it should be changed to `const opportunityManager = new OpportunityManager();`

File: tests/services/opportunity-test.js
Comment (Manvityagi) [inline tests/services/opportunity-test.js]: `Opprtunity` spelling

**Review by Manvityagi — CHANGES_REQUESTED** [review]
Other than the little things in review comments, the rest looks good to me. 

PS: I see some of the files are not properly formatted, use Shift+Alt+F for formatting before pushing.

**Review by Manvityagi — APPROVED** [review]
Looks good to me @Sukriti-sood 

Waiting for @abdus approval for merging

**Review by abdus — APPROVED** [review]

**Comment by Manvityagi** [comment]
Absolutely good work and effort @Sukriti-sood. Merged finally :)

**Comment by Sukriti-sood** [comment]
> Absolutely good work and effort @Sukriti-sood. Merged finally :)

Thanks @Manvityagi  , Would love to contribute more in Backend of this Project with more API's and Testing.

**Comment by Manvityagi** [comment]
> > Absolutely good work and effort @Sukriti-sood. Merged finally :)
> 
> Thanks @Manvityagi , Would love to contribute more in Backend of this Project with more API's and Testing.

Sure, There are some freshly opened issues, Check them out.

**Pull request merged**
